app/src/utils/utils.ts

```diff
@@ -161,3 +161,77 @@ export async function assertSufficientBalance(tokenAddress: string, requiredAmou
   }
   return true;
 }
+
+/**
+ * @notice Get the multicall data for the fns on a contract @ address
+ * @param calls {target, contract, fns} The address, contract and fns we want to call via multicall
+ * @returns A list of calls we can make to multicall.tryBlockAndAggregate()
+ */
+export const getMulticallData = (
+  calls: {
+    target: string;
+    contract: Contract;
+    fns: (string | { fn: string; args: (string | BigNumberish | boolean)[] })[];
+  }[]
+) => {
+  const flatMap: {
+    target: string;
+    callData: string;
+  }[] = [];
+  calls.forEach((call) =>
+    flatMap.push(
+      ...call.fns.map((fn) => {
+        const args = typeof fn === 'string' ? [fn] : [fn.fn, fn.args];
+        return {
+          target: call.target,
+          callData: call.contract.interface.encodeFunctionData(
+            args[0] as string,
+            args[1] as (string | BigNumberish | boolean)[]
+          ),
```

Comment:
Does `getMulticallData` only calls that have up to two args of these specified types?